submissions/RomanSaiko/tiny-js-world/index.js

```diff
@@ -0,0 +1,57 @@
+const dog = {
+    species: 'dog',
+    name: 'Spike',
+    gender: 'male',
+    legs: 4,
+    hands: 0,
+    saying: 'woof-woof',
+    friends: ['Dolf', 'Roman', 'Mariia']
+}
+
+const cat = {
+    species: 'cat',
+    name: 'Dolf',
+    gender: 'male',
+    legs: 4,
+    hands: 0,
+    saying: 'meow-meow',
+    friends: ['Anne Hathaway']
+}
+
+const man = {
+    species: 'man',
+    name: 'Roman',
+    gender: 'male',
+    legs: 2,
+    hands: 2,
+    saying: 'Hello!',
+    friends: ['Dolf', 'Spike', 'Mariia']
+}
+
+const woman = {
+    species: 'woman',
+    name: 'Mariia',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    saying: 'Hi!',
+    friends: ['Dolf', 'Spike', 'Roman']
+}
+
+const catWoman = {
+    species: 'cat-woman',
+    name: 'Anne Hathaway',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    saying: cat.saying,
+    friends: ['Dolf', 'Spike']
+}
+
+const inhabitants = [dog, cat, man, woman, catWoman]
+
+console.log(inhabitants);
+
+for(let i = 0; i < inhabitants.length; i++) {
+    print(`${inhabitants[i].species}; ${inhabitants[i].name}; ${inhabitants[i].gender}; ${inhabitants[i].legs}; ${inhabitants[i].hands}; ${inhabitants[i].saying}; ${inhabitants[i].friends.map(item => item)}.`);
+}
```

Comment:
Please employ an appropriate `Array` method to iterate through list items.

Please encapsulate the output string creation in a function. In that function you might also list object properties and an `Array` method to convert them into values. Of course `freinds` property would still require specific handling.

Reply:
Too much code repetition. Keep code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) as much as possible. This is a tiny task and perhaps this effort is a bit tto much for it, but you will find the approach paying on bigger tasks and at work.

What if you are required to add another 10 properties? With current approach you will need to add each twice, in decomposition and in template string literal. With a DRY approach you have a single list of property names.

Hint:
`['propA', 'propB', 'propC'].map(propName => obj[propName])` results in an array of object property values.

Repetetive `; ` can be replaced with a proper `.join()`

Yet `friends` will require special handling, true.

Combination of `.map`'s, `.join`'s, `.forEach`'s, sometimes nested will do the magic.